integration test `test_results_correctness` fails intermittently

the error message looks something like this:

``` 
F
test_results_correctness.py:33 (test_similarity_scores)
requests = <requests.sessions.Session object at 0x10e3efd30>
the_truth = {'clks_a': ['FttXOHbv3khjweS35so7Jva8W3ef5uK5+HSuNPUsYNmihlvhn7QhPFkUfvHmzfw+Ycc3Mv4zdxlLaiDTG4rFvAOa7mtAL+p3TVQy3pLwT...['0', '1', '10', '100', '1000', '1001', ...], 'entity_ids_b': ['189', '1534', '1478', '182', '1272', '1794', ...], ...}

    def test_similarity_scores(requests, the_truth):
        project_data, _, _ = create_project_upload_data(requests, the_truth['clks_a'], the_truth['clks_b'],
                                                        result_type='similarity_scores')
        run = post_run(requests, project_data, threshold=the_truth['threshold'])
        result = get_run_result(requests, project_data, run)
        # compare the result with the truth
        ss = result['similarity_scores']
        ts = the_truth['similarity_scores']
        assert len(ss) == len(ts)
        for es_score, true_score in zip(ss, ts):
>           assert es_score[0] == true_score[0] and es_score[1] == true_score[2]
E           assert (0 == 0 and 990 == 818)

test_results_correctness.py:44: AssertionError
```

--- a/backend/entityservice/database/selections.py
+++ b/backend/entityservice/database/selections.py
@@ -137,6 +137,7 @@ def get_project_dataset_sizes(db, project_id):
         WHERE
           bloomingdata.dp=dataproviders.id AND
           dataproviders.project=%s
+        ORDER BY dataproviders.id
         """
     query_result = query_db(db, sql_query, [project_id], one=False)
     return [r['size'] for r in query_result]

